src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -1066,8 +1079,8 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
     SetFullHealth();
     SetPower(POWER_MANA, GetMaxPower(POWER_MANA));
 
-    if (Pet* pet = ToPet())
-        sScriptMgr->OnPetInitStatsForLevel(pet);
+    if (owner->GetTypeId() == TYPEID_PLAYER)
+        sScriptMgr->OnAfterGuardianInitStatsForLevel(owner->ToPlayer(), this);
```

Comment:
this can break other modules/scripts. It's good to refactor it but maybe we need to offer a "deprecation/legacy" system for hooks. Maybe we can put the deprecated calls inside the "OnAfterGuardianInitStatsForLevel" and put an sLog about the deprecation inside those hooks.
What do you think about it?

Reply:
This won't break anything because I was the one introducing the hook "OnPetInitStatsForLevel" in this PR: #2556
I needed it for my mod-morphsummon module, so I'm the only one using it at the moment (I'll update the module accordingly).